context/context.go

```diff
@@ -202,3 +256,60 @@ func (c *ctx) returnToPool() {
 	c.Reset()
 	c.pool.Put(c)
 }
+
+func (c *ctx) NewChildContext() Context {
+	var child Context
+	if c.pool == nil {
+		child = NewContext()
+	} else {
+		child = c.pool.Get()
+	}
+
+	child.SetParentCtx(c)
+
+	return child
+}
+
+func (c *ctx) SetParentCtx(parentCtx Context) {
+	// check to see if parent exists?
+	c.parent = parentCtx
+}
+
+func (c *ctx) ParentCtx() Context {
```

Comment:
I think you should pick one of:
```
NewChildContext()
ParentContext()
```

or
```
NewChild()
Parent()
```

For consistency.  Also I would optimize the locking here to avoid defer.

Reply:
Actually on second thoughts, let's just unexport all these methods - we only need to add StartSpan for now.